File: src/main/java/no/nav/melosys/eessi/controller/dto/Lovvalgsperiode.java
package no.nav.melosys.eessi.controller.dto;

import java.time.LocalDate;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonFormat.Shape;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.Data;

@Data
public class Lovvalgsperiode {

    private String lovvalgsland;
    private String unntakFraLovvalgsland;
    private Bestemmelse bestemmelse;

    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonFormat(shape = Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate fom;

    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonFormat(shape = Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate tom;

    private String unntaksBegrunnelse;

    private Bestemmelse unntakFraBestemmelse;
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Lovvalgsperiode.kt
package no.nav.melosys.eessi.controller.dto

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonFormat.Shape
import java.time.LocalDate

data class Lovvalgsperiode(
    var lovvalgsland: String? = null,
    var unntakFraLovvalgsland: String? = null,
    var bestemmelse: Bestemmelse? = null,
    var tilleggsBestemmelse: Bestemmelse? = null,

    @JsonFormat(shape = Shape.STRING, pattern = "yyyy-MM-dd")
    var fom: LocalDate? = null,

    @JsonFormat(shape = Shape.STRING, pattern = "yyyy-MM-dd")
    var tom: LocalDate? = null,

    var unntaksBegrunnelse: String? = null,
    var unntakFraBestemmelse: Bestemmelse?  = null
) {
    fun harTilleggsbestemmelse() = tilleggsBestemmelse != null
}
